pkg/cache/cache.go

```diff
@@ -1,173 +1,43 @@
-			return b, true
-		}
-	}
-	return nil, false
-}
-
-// MultiGet can be used to fetch several keys at once.
-func (c Cache) MultiGet(keys []string) [][]byte {
-	results := make([][]byte, len(keys))
-	if c.client == nil {
-		for i, key := range keys {
-			results[i], _ = c.Get(key)
-		}
-	} else {
-		cmd := c.client.MGet(c.ctx, keys...)
-		for i, val := range cmd.Val() {
-			if buf, ok := val.(string); ok {
-				results[i] = []byte(buf)
-			}
-		}
-	}
-	return results
+type Cache interface {
+	CheckStatus(ctx context.Context) (time.Duration, error)
+	Get(key string) ([]byte, bool)
+	MultiGet(keys []string) [][]byte
+	Keys(prefix string) []string
+	Clear(key string)
+	Set(key string, data []byte, expiration time.Duration)
+	SetNX(key string, data []byte, expiration time.Duration)
+	GetCompressed(key string) (io.Reader, bool)
+	SetCompressed(key string, data []byte, expiration time.Duration)
+	RefreshTTL(key string, expiration time.Duration)
 }
 
-// Keys returns the list of keys with the given prefix.
-// Note: it can be slow and should be used carefully.
-func (c Cache) Keys(prefix string) []string {
-	if c.client != nil {
-		cmd := c.client.Keys(c.ctx, prefix+"*")
-		return cmd.Val()
-	}
-	results := make([]string, 0)
-	c.m.Range(func(key, _ interface{}) bool {
-		k := key.(string)
-		if strings.HasPrefix(k, prefix) {
-			results = append(results, k)
-		}
-		return true
-	})
-	return results
-}
-
-// Clear removes a key from the cache
-func (c Cache) Clear(key string) {
-	if c.client == nil {
-		c.m.Delete(key)
-	} else {
-		c.client.Del(c.ctx, key)
-	}
-}
-
-// Set stores an asset to the given key.
-func (c Cache) Set(key string, data []byte, expiration time.Duration) {
-	if c.client == nil {
-		c.m.Store(key, cacheEntry{
-			payload:   data,
-			expiredAt: time.Now().Add(expiration),
-		})
-	} else {
-		c.client.Set(c.ctx, key, data, expiration)
-	}
-}
-
-// SetNX stores the data in the cache only if the key doesn't exist yet.
-func (c Cache) SetNX(key string, data []byte, expiration time.Duration) {
-	if c.client == nil {
-		c.m.LoadOrStore(key, cacheEntry{
-			payload:   data,
-			expiredAt: time.Now().Add(expiration),
-		})
-	} else {
-		c.client.SetNX(c.ctx, key, data, expiration)
-	}
-}
-
-// GetCompressed works like Get but expect a compressed asset that is
-// uncompressed.
-func (c Cache) GetCompressed(key string) (io.Reader, bool) {
-	if r, ok := c.Get(key); ok {
-		if gr, err := gzip.NewReader(bytes.NewReader(r)); err == nil {
-			return gr, true
-		}
-	}
-	return nil, false
+type cacheEntry struct {
+	payload   []byte
+	expiredAt time.Time
 }
 
-// SetCompressed works like Set but compress the asset data before storing it.
-func (c Cache) SetCompressed(key string, data []byte, expiration time.Duration) {
-	dataCompressed := new(bytes.Buffer)
-	gw := gzip.NewWriter(dataCompressed)
-	defer gw.Close()
-	if _, err := io.Copy(gw, bytes.NewReader(data)); err != nil {
-		return
+// Init instantiate a Cache Client.
```

Comment:
I would prefer to call this function `New`. For me, `Init` is called for its side effect on the package, while `New` is called to return an object.